packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -717,5 +718,182 @@ describe('build-schema', () => {
         expect(schema).to.deepEqual(expectedSchemaForCategory);
       });
     });
+
+    it('converts HasMany and BelongsTo relation links', () => {
+      @model()
+      class Product extends Entity {
+        @property({id: true})
+        id: number;
+
+        @belongsTo(() => Category)
+        categoryId: number;
+      }
+
+      @model()
+      class Category extends Entity {
+        @property({id: true})
+        id: number;
+
+        @hasMany(() => Product)
+        products?: Product[];
+      }
+
+      @model()
+      class CategoryWithoutProp extends Entity {
+        @hasMany(() => Product)
+        products?: Product[];
+      }
+      const expectedSchema: JsonSchema = {
+        definitions: {
+          ProductWithRelations: {
+            title: 'ProductWithRelations',
+            properties: {
+              id: {type: 'number'},
+              categoryId: {type: 'number'},
+              category: {$ref: '#/definitions/CategoryWithRelations'},
+            },
+          },
+        },
+        properties: {
+          id: {type: 'number'},
+          products: {
+            type: 'array',
+            items: {$ref: '#/definitions/ProductWithRelations'},
+          },
+        },
+        title: 'CategoryWithRelations',
```

Comment:
Can you list the schema fields in the order `title`, `properties`, `definitions` please? I find it difficult to read `expectedScheme` when the order is reversed.

Reply:
I had to order it like that because the getJsonSchema method is returning the data in that order and if I don't keep it like that, the test fails. Is there any other way out here ?

Reply:
I see. Let's not worry about this too much, we can keep the current form.